cscs-checks/tools/profiling_and_debugging/intel_advisor_roofline.py

```diff
@@ -0,0 +1,161 @@
+        ScalarAddbw = 3*1024**3
+        self.sanity_patterns = sn.all([
+            # check the job status:
+            sn.assert_found('loop complete.', self.stdout),
+            # check the tool's version:
+            sn.assert_eq(sn.extractsingle(
+                r'I*.\(build\s(?P<toolsversion>\d+)\s*.',
+                self.version_rpt, 'toolsversion'), toolsversion),
+            # --- roofline boundaries:
+            # check --report=roofs (L1 bandwidth):
+            sn.assert_reference(sn.extractsingle(
+                r'^L1\sBandwidth\s\(single-threaded\)\s+(?P<L1bw>\d+)\s+'
+                r'memory$', self.roofline_ref, 'L1bw', int),
+                L1bw, -0.08, 0.08),
+            # check --report=roofs (L2 bandwidth):
+            sn.assert_reference(sn.extractsingle(
+                r'^L2\sBandwidth\s\(single-threaded\)\s+(?P<L2bw>\d+)\s+'
+                r'memory$', self.roofline_ref, 'L2bw', int),
+                L2bw, -0.08, 0.08),
+            # check --report=roofs (L3 bandwidth):
+            sn.assert_reference(sn.extractsingle(
+                r'^L3\sBandwidth\s\(single-threaded\)\s+(?P<L3bw>\d+)\s+'
+                r'memory$', self.roofline_ref, 'L3bw', int),
+                L3bw, -0.08, 0.08),
+            # check --report=roofs (DP FMA):
+            sn.assert_reference(sn.extractsingle(
+                r'^DP Vector FMA Peak\s\(single-threaded\)\s+'
+                r'(?P<DPfmabw>\d+)\s+compute$', self.roofline_ref,
+                'DPfmabw', int), DPfmabw, -0.08, 0.08),
+            # check --report=roofs (DP Add):
+            sn.assert_reference(sn.extractsingle(
+                r'^DP Vector Add Peak\s\(single-threaded\)\s+'
+                r'(?P<DPaddbw>\d+)\s+compute$', self.roofline_ref,
+                'DPaddbw', int), DPaddbw, -0.08, 0.08),
+            # check --report=roofs (Scalar Add):
+            sn.assert_reference(sn.extractsingle(
+                r'^Scalar Add Peak\s\(single-threaded\)\s+'
+                r'(?P<ScalarAddbw>\d+)\s+compute$', self.roofline_ref,
+                'ScalarAddbw', int), ScalarAddbw, -0.08, 0.08),
+            # --- check Arithmetic_intensity:
+            sn.assert_reference(sn.extractsingle(
+                r'^returned\sAI\sgap\s=\s(?P<Intensity>.*)', self.roofline_rpt,
+                'Intensity', float), 0.0, -0.01, 0.01),
+            # --- check GFLOPS:
+            sn.assert_reference(sn.extractsingle(
+                r'^returned\sGFLOPS\sgap\s=\s(?P<Flops>.*)', self.roofline_rpt,
+                'Flops', float), 0.0, -0.01, 0.01),
+        ])
+        self.maintainers = ['JG']
+        self.tags = {'production'}
+
+    def setup(self, partition, environ, **job_opts):
+        super().setup(partition, environ, **job_opts)
+        environ_name = self.current_environ.name
+        prgenv_flags = self.prgenv_flags[environ_name]
+        self.build_system.cxxflags = prgenv_flags
+        launcher_cmd = ''.join(self.job.launcher.command(self.job))
```

Comment:
Since it returns you a list of arguments, you should join with spaces:

```suggestion
        launcher_cmd = ' '.join(self.job.launcher.command(self.job))
```

Reply:
:ok: